app/assets/javascripts/discourse/app/lib/sidebar/user/categories-section/category-section-link.js

```diff
@@ -1,35 +1,121 @@
 import I18n from "I18n";
 
 import { tracked } from "@glimmer/tracking";
+import { get, set } from "@ember/object";
 
 import { bind } from "discourse-common/utils/decorators";
 import Category from "discourse/models/category";
 import { UNREAD_LIST_DESTINATION } from "discourse/controllers/preferences/sidebar";
 
+const DEFAULT_COUNTABLES = [
+  {
+    propertyName: "totalUnread",
+    badgeTextFunction: (count) => {
+      return I18n.t("sidebar.unread_count", { count });
+    },
+    route: "discovery.unreadCategory",
+    refreshCountFunction: ({ topicTrackingState, category }) => {
+      return topicTrackingState.countUnread({
+        categoryId: category.id,
+      });
+    },
+  },
+  {
+    propertyName: "totalNew",
+    badgeTextFunction: (count) => {
+      return I18n.t("sidebar.new_count", { count });
+    },
+    route: "discovery.newCategory",
+    refreshCountFunction: ({ topicTrackingState, category }) => {
+      return topicTrackingState.countNew({
+        categoryId: category.id,
+      });
+    },
+  },
+];
+
+const customCountables = [];
+
+export function registerCustomCountable({
+  badgeTextFunction,
+  route,
+  routeQuery,
+  shouldRegister,
+  refreshCountFunction,
+  prioritizeOverDefaults,
+}) {
+  const length = customCountables.length + 1;
+
+  customCountables.push({
+    propertyName: `customCountableProperty${length}`,
+    badgeTextFunction,
+    route,
+    routeQuery,
+    shouldRegister,
+    refreshCountFunction,
+    prioritizeOverDefaults,
+  });
+}
+
+export function resetCustomCountables() {
+  customCountables.length = 0;
```

Comment:
This seems a little odd to me, can't we simply use `.clear()` here?

```suggestion
  customCountables.clear();
```

Reply:
Where does the `clear()` API come from? I couldn't find that in the docs anywhere.

Reply:
Ah sorry, forgot to reply. I thought `clear()` was a native method of arrays, but it's actually added by Ember to the Array prototype: https://api.emberjs.com/ember/3.5/classes/MutableArray/methods/arrayContentWillChange?anchor=clear.

Anyways, it's super minor so I'm approving.